front/lib/api/assistant/email_trigger.ts

```diff
@@ -307,14 +309,37 @@ export async function triggerFromEmail({
   // console.log("REST_OF_THREAD", restOfThread, restOfThread.length);
 
   if (restOfThread.length > 0) {
+    const file = await FileResource.makeNew({
+      contentType: "text/plain",
+      fileName: `email-${email.subject}.txt`,
+      fileSize: restOfThread.length,
+      userId: user.id,
+      workspaceId: auth.getNonNullableWorkspace().id,
+      useCase: "conversation",
+      useCaseMetadata: null,
+    });
+
+    const processRes = await processAndStoreFile(auth, {
```

Comment:
FYI, we may want to align what we do here & in `front/lib/api/files/tool_output.ts`

Reply:
I think the approach here is more general no as we don't hard code the processing nature?

Reply:
Agree, do you think you oculd move the existing code to this pattern ?

Reply:
Will align the two :+1:

Reply:
Any chance we can create a generic helper for this in lib/api/files? I feel like we keep duplicating:

```
const file = await FileResource.makeNew();

...

const processRes = await processAndStoreFile(auth, {});
```